File: springboot/src/main/java/com/project/Timetable/models/Session.java
package com.project.Timetable.models;

import com.project.Timetable.models.Enum.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Session {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    @Column(nullable = false)
    private LocalDate date;

    @Enumerated(EnumType.STRING)
    private Status status;

    @ManyToOne
    @JoinColumn(name = "timeslot_id")
    private TimeSlot timeSlot;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @ManyToOne
    @JoinColumn(name = "users_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "group_id")
    private Group group;

    @ManyToOne
    @JoinColumn(name = "course_id")
    private Course course;

    @OneToMany(mappedBy = "session")
    private List<Attendance> attendance;
}

File: angular/src/app/models/Session.ts
import { Attendance } from './Attendance';
import { Course } from './Course';
import { Group } from './Group';
import { Room } from './Room';
import { TimeSlot } from './TimeSlot';
import { User } from './User';

export class Session {
  id!: number;
  date!: Date;
  status!: string;
  course?: Course;
  group?: Group;
  room?: Room;
  timeslot?: TimeSlot;
  user?: User;
  attendances?: Attendance[];
}
